bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/lib/GC100IRControlPoint.java

```diff
@@ -0,0 +1,582 @@
+    private String responseOfDevice(String response) {
+        if (response.equals("unknowncommand, 1")) {
+            return ("Time out occurred because carriage return <CR> not received. The request was not processed.");
+        } else if (response.equals("unknowncommand, 2")) {
+            return ("Invalid module address (module does not exist) received when attempting to ascertain the version number (getversion).");
+        } else if (response.equals("unknowncommand, 3")) {
+            return ("Invalid module address (module does not exist).");
+        } else if (response.equals("unknowncommand, 4")) {
+            return ("Invalid connector address.");
+        } else if (response.equals("unknowncommand, 5")) {
+            return ("Connector address 1 is set up as �sensor in� when attempting to send an IR command.");
+        } else if (response.equals("unknowncommand, 6")) {
+            return ("Connector address 2 is set up as �sensor in� when attempting to send an IR command.");
+        } else if (response.equals("unknowncommand, 7")) {
+            return ("Connector address 3 is set up as �sensor in� when attempting to send an IR command.");
+        } else if (response.equals("unknowncommand, 8")) {
+            return ("Offset is set to an even transition number, but should be set to an odd transition number in the IR command.");
+        } else if (response.equals("unknowncommand, 9")) {
+            return ("Maximum number of transitions exceeded (256 total on/off transitions allowed).");
+        } else if (response.equals("unknowncommand, 10")) {
+            return ("Number of transitions in the IR command is not even (the same number of on and off transitions is required).");
+        } else if (response.equals("unknowncommand, 11")) {
+            return ("Contact closure command sent to a module that is not a relay..");
+        } else if (response.equals("unknowncommand, 12")) {
+            return ("Missing carriage return. All commands must end with a carriage return.");
+        } else if (response.equals("unknowncommand, 13")) {
+            return ("State was requested of an invalid connector address, or the connector is programmed as IR out and not sensor in.");
+        } else if (response.equals("unknowncommand, 14")) {
+            return ("Command sent to the unit is not supported by the GC-100.");
+        } else if (response.equals("unknowncommand, 15")) {
+            return ("Maximum number of IR transitions exceeded. (SM_IR_INPROCESS)");
+        } else if (response.equals("unknowncommand, 16")) {
+            return ("Invalid number of IR transitions (must be an even number).");
+        } else if (response.equals("unknowncommand, 21")) {
+            return ("Attempted to send an IR command to a non-IR module.");
+        } else if (response.equals("unknowncommand, 23")) {
+            return ("Command sent is not supported by this type of module.");
+        }
+
+        return response;
+    }
```

Comment:
this method would be better if it did a lookup in a `private static final Map<String,String> responses;` static member, and return `response` if not in the map.